### PR TITLE
Fix leading trivia when rewriting field builder

### PR DESCRIPTION
**Pull request by gao-artur**

### CODE REVIEW

**Comment by gao-artur** [comment]
Strange. The `CarriageReturnLineFeed` used to work in both OS before this change.

**Comment by Shane32** [comment]
Maybe `Environment.NewLine` ?

**Comment by Shane32** [comment]
Another thing to consider is that Git may decode line feeds differently on Linux vs Windows.  So if there's a text literal within the code like this: `"Test\nTest2"`, that will be the same on both operating systems, but when you write `@"Test` with `Test2"` on the next line, then the literal itself might change based on the OS it was compiled on.  Just something to keep in mind when writing tests.

File: src/GraphQL.Analyzers.CodeFixes/FieldBuilderCodeFixProvider.cs [hunk 1]
```diff
@@ -77,6 +77,17 @@ public override async Task RegisterCodeFixesAsync(CodeFixContext context)
...
+        const int oneLevelIndentation = 4;
+        int fieldInvocationIndentation = newFieldInvocationExpression
+            .GetLeadingTrivia()
+            .Where(trivia => trivia.IsKind(SyntaxKind.WhitespaceTrivia))
+            .Sum(trivia => trivia.FullSpan.Length);
+
+        var whitespaceTrivia = Whitespace(new string(' ', fieldInvocationIndentation + oneLevelIndentation));
+        var newLineTrivia = RuntimeInformation.IsOSPlatform(OSPlatform.Windows)
+            ? CarriageReturnLineFeed
+            : LineFeed;
```
Comment (Shane32) [inline src/GraphQL.Analyzers.CodeFixes/FieldBuilderCodeFixProvider.cs:89]: Is this not equivalent to `Environment.NewLine` ?

https://learn.microsoft.com/en-us/dotnet/api/system.environment.newline?view=net-7.0

> `\r\n` for non-Unix platforms, or `\n` for Unix platforms.

Comment (gao-artur) [reply src/GraphQL.Analyzers.CodeFixes/FieldBuilderCodeFixProvider.cs:89]: Will try this
```c#
EndOfLine(Environment.NewLine)
```

File: src/GraphQL.Analyzers.CodeFixes/FieldBuilderCodeFixProvider.cs [hunk 2]
```diff
@@ -236,18 +246,16 @@ static SyntaxToken GetIdentifier(SimpleNameSyntax simpleNameSyntax, string name)
...
                             Argument(argumentExpression))));
 
         return exp;
     }
-
-    private static SyntaxNode Format(SyntaxNode syntaxNode, Workspace workspace)
-    {
-        syntaxNode = syntaxNode.WithAdditionalAnnotations(Formatter.Annotation);
-        return Formatter.Format(syntaxNode, Formatter.Annotation, workspace);
-    }
```
Comment (gao-artur) [inline src/GraphQL.Analyzers.CodeFixes/FieldBuilderCodeFixProvider.cs:252]: This method was normalizing the line endings. But it also makes wrong indentations.

Comment (Shane32) [reply src/GraphQL.Analyzers.CodeFixes/FieldBuilderCodeFixProvider.cs:252]: Do we have a new test added for the scenario fixed by this PR?

Comment (gao-artur) [reply src/GraphQL.Analyzers.CodeFixes/FieldBuilderCodeFixProvider.cs:252]: I have adjusted an old test. This is the result before the fix
image

Comment (Shane32) [reply src/GraphQL.Analyzers.CodeFixes/FieldBuilderCodeFixProvider.cs:252]: oh i see.  :+1:

**Review by Shane32 — APPROVED** [review]

**Comment by Shane32** [comment]
All set to merge @gao-artur  ?

**Comment by gao-artur** [comment]
Yep

**Comment by Shane32** [comment]
Available in `GraphQL.7.6.2-preview-859`

**Pull request merged**